### PR TITLE
Promise.all polyfill

### PR DESCRIPTION
**Pull request by dvoytenko**
/cc @cramforce

### CODE REVIEW

**Comment by erwinmombay** [comment]
LGTM

File: test/functional/test-promise.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    return promise.all(promises).then(results => {
+      expect(results).to.have.length.of(3);
+      expect(results[0]).to.equal('A');
+      expect(results[1]).to.equal('B');
+      expect(results[2]).to.equal('C');
+    });
+  });
+
+  it('should fail as soon as one fails', () => {
```
Comment (cramforce) [inline test/functional/test-promise.js]: Is this correct?

Comment (cramforce) [inline test/functional/test-promise.js]: Just confirming. I think it is.

**Comment by cramforce** [comment]
LGTM

File: test/functional/test-promise.js (hunk 1 above)
Comment (dvoytenko) [inline test/functional/test-promise.js]: Yes. Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

> If any of the passed in promises rejects, the all Promise immediately rejects with the value of the promise that rejected, discarding all the other promises whether or not they have resolved.

Comment (erwinmombay) [inline test/functional/test-promise.js]: seems to be:

Promise.all fail-fast behaviour

Promise.all is rejected if one of the elements is rejected and Promise.all fails fast: If you have four promises which resolve after a timeout, and one rejects immediately, then Promise.all rejects immediately.

**Pull request merged**